### PR TITLE
Beat That Submission

### PR DESCRIPTION
**Pull request by kangraye**
Please fill out the survey before submitting the pull request. Thanks!

🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀🚀

How many hours did you spend on this assignment? 
4
Please fill in one error and/or error message you received while working on this assignment. 
Nil.
What part of the assignment did you spend the most time on?
Figuring out the scoreboard and fixtures.
Comfort Level (1-5):
5
Completeness Level (1-5):
5
What did you think of this deliverable?
Might have some bugs given the extensiveness but I think it should work.
Is there anything in this code that you feel pleased about?
Completed all more comfortable components.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -73,7 +75,7 @@ <h1 id="header">Basics: Beat That! 🚀</h1>
         output.innerHTML = result;
 
         // Reset input value
-        input.value = "";
+        document.querySelector("#title-text").innerHTML = textUpdate();
```
Comment (emily-zhangg) [inline index.html:78]: great job using JS to manipulate HTML DOM elements!

File: script.js [hunk 2]
```diff
@@ -1,4 +1,421 @@
...
+  if (input == LOW || input == HIGH) {
+    lowHigh = input;
+    outputValue = `You have selected that the ${lowHigh}EST roll wins.`;
+    gameState = AUTO_CHOOSE;
+    return outputValue;
+  } else {
+    outputValue = `Invalid entry. Please input either ${LOW} or ${HIGH}.`;
+    return outputValue;
+  }
+};
```
Comment (emily-zhangg) [inline script.js:122]: I like the use of helper functions and comments to make your code much more organized and readable

File: script.js [hunk 3]
```diff
@@ -1,4 +1,421 @@
...
+
+// Helper Function to Combine Integers in Array
+var combineArray = function (array) {
+  var arrayLength = array.length;
+  var outputValue = 0;
+  for (var counter = 0; counter < arrayLength; counter += 1) {
+    outputValue = 10 * outputValue + array[counter];
+  }
+  return outputValue;
```
Comment (emily-zhangg) [inline script.js:131]: good job implementing for loops here

File: script.js [hunk 4]
```diff
@@ -1,4 +1,421 @@
...
+};
+
+// Helper Function to Generate Fixtures into Global Variables currentFixtures and nextFixtures
+var generateFixtures = function () {
+  var playerList = Array(numPlayers)
+    .fill()
+    .map((_, i) => i + 1);
+  var randomizedFixture = playerList.sort(() => Math.random() - 0.5);
+  // Check how many players get a pass to the next round
+  var numberOfPasses = 2 ** Math.ceil(Math.log2(numPlayers)) - numPlayers;
```
Comment (emily-zhangg) [inline script.js:180]: I like how you managed to implement the array methods you found online in your own code

**Review by yiqiingg — COMMENTED** [review]
really great job implementing this project. I like how you manage to take the feedback from the previous project and made your code more organized and readable. Great work using the internet to learn more about the different methods you can call on arrays and implementing them in your work.

**Pull request closed without merging**